### PR TITLE
Enable publishing metrics to Prometheus

### PR DESCRIPTION
**Pull request by chetanmeh**
Enables support for exporting the metrics to [Prometheus][1] via [Kamon Prometheus][2] exporter.

## Description

This PR enables support for exposing the metrics via `/metrics` web endpoint. By default the metrics are not exposed. It can be enabled via following config 

```
whisk {
    metrics {
        prometheus-enabled = false
    }
}
```

Sample output of `/metrics` endpoint

```
# TYPE counter_loadbalancer_invokerState_unhealthy_count_total counter
counter_loadbalancer_invokerState_unhealthy_count_total 0.0
# TYPE counter_controller_startup0_count_total counter
counter_controller_startup0_count_total 0.0
# TYPE counter_database_getDocument_start_total counter
counter_database_getDocument_start_total 0.0
# TYPE counter_loadbalancer_activations_count_total counter
counter_loadbalancer_activations_count_total 0.0
# TYPE counter_database_saveDocument_start_total counter
counter_database_saveDocument_start_total 0.0
# TYPE counter_controller_kafka_start_total counter
counter_controller_kafka_start_total 0.0
# TYPE counter_http_get_200_count_total counter
counter_http_get_200_count_total 1.0
# TYPE counter_database_cacheMiss_count_total counter
counter_database_cacheMiss_count_total 0.0
```

It needs the Kamon integration to be enabled. This PR also moves some of the config option from env variables to typesafe config (but still supports env based config for compatibility)

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [x] Deployment
- [ ] CLI
- [x] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [ ] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [ ] I signed an [Apache CLA](https://github.com/apache/incubator-openwhisk/blob/master/CONTRIBUTING.md).
- [ ] I reviewed the [style guides](https://github.com/apache/incubator-openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [ ] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

[1]: https://prometheus.io/
[2]: https://github.com/kamon-io/kamon-prometheus

### CODE REVIEW

**Comment by tysonnorris** [comment]
@vvraskin is PG good?

**Comment by rabbah** [comment]
@chetanmeh looks like a conflict on WhiskConfig - can you fix then we can merge.

File: tests/src/test/scala/org/apache/openwhisk/common/PrometheusTests.scala [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      withClue(responseText) {
+        responseText should include("foo_bar")
+      }
+    }
+    api.close()
+  }
+
+  it should "not be enabled by default" in {
+    Get("/metrics") ~> MetricsRoute() ~> check {
+      handled shouldBe false
```
Comment (rabbah) [inline tests/src/test/scala/org/apache/openwhisk/common/PrometheusTests.scala]: will subsequent PRs make this route available via configuration?

**Comment by chetanmeh** [comment]
Idea here was to have it disabled by default and enable it via config (see PR description on config details)

Currently we do not have a good way to document such config as most docs are around documenting the ansible flags. So currently it's documented in conf file itself

**Comment by vvraskin** [comment]
@chetanmeh could you please resolve the conflict, I will give it another PG round and merge?

**Comment by chetanmeh** [comment]
rebased now

**Comment by vvraskin** [comment]
I've updated our downstream with necessary changes and rerunning a PG now. Will get back once its ready.

**Comment by vvraskin** [comment]
PG is 👍 , will be able to merge it once the conflicts are resolved.

**Comment by rabbah** [comment]
I rebased this again to master.

**Comment by CPattanayak** [comment]
Hi please letme know is it updated in kubernet chart i need to get this metrics in kubernet

**Pull request merged**